README.md

```diff
@@ -1 +1,241 @@
+  Id int64
+
+  // This is the `clientName` part of the client certificate CN
+  // Used to distinguish who created this job
+  ClientName string
+  Command string
+  StartTime date
+  // EndTime is a date pointer to allow it to be nil
+  EndTime *date
+  // ExitCode is a int32 pointer to allow it to be nil
+  ExitCode *int32
+
+  Logs []string
+  mu sync.Mutex
+}
+```
+
+Each job needs a mutex so that the goroutine that can is running the job can periodically update it with stored logs. When a client connects (via the `logs` command), then can get the full output of this slice, followed by either periodic updates (when the slice length changes), or subscribe to the job runner that's actively running the job.
+
+Job starts need to be handled by an object that is thread safe, for consistency with creating jobIds. Lowering coordination requirements here may be provide a performance improvement, but that is currently not in scope.
+
+```go
+
+type JobRunner struct {
+  jobs map[int64]*Job
+  jobsByClient map[string]*Job
+  mu sync.Mutex
+}
+
+func (j *JobRunner) Start(command, client string) *JobRunner {
+  j.mu.Lock()
+  defer j.mu.Unlock()
+
+  // do stuff
+}
+```
+
+### CLI
+
+```text
+pjw list          Provides a list of jobs and their status
+pjw get ID        Provides a detailed status of a specific job
+pjw start CMD     Create and start a job; returns the job id
+pjw stop ID       Stop a job. Returns success if the job is already stopped/done
+pjw logs ID       Stream the logs of a job, always starting at the beginning
+```
+
+### Security
+
+#### Authentication
+
+Authentication is based on mTLS and is also completely manual to setup. The server needs a Certificate Authority (CA) and public and private server certificates. It does not generate any of this on it's own. Securely storing and presenting the CA and private server certs is the responsibility of the application operator.
+
+Examples for creating certs:
+
+```shell
+DAYS=20
+SUBJECT="/C=US/ST=Washington/L=Seattle/O=GhostSquad/OU=Prototype Job Worker/CN=localhost"
+
+# CA
+openssl req -newkey rsa:4096 \
+  -new -nodes -x509 \
+  -days "${DAYS}" \
+  -out ca.crt \
+  -keyout ca.key \
+  -subj "${SUBJECT}"
+
+# Server
+openssl genrsa -out server.key 4096
+openssl req -new -key server.key -days "${DAYS}" -out server.csr \
+    -subj "${SUBJECT}"
+
+openssl x509  -req -in server.csr \
+    -extfile <(printf "subjectAltName=DNS:localhost") \
+    -CA ca.crt -CAkey ca.key  \
+    -days "${DAYS}" -sha256 -CAcreateserial \
+    -out server.crt
+
+# Client
+CLIENT=john
+
+openssl genrsa -out "${CLIENT}.key" 4096
+
+openssl req -new -key "${CLIENT}.key" -days "${DAYS}" -out "${CLIENT}.csr" \
+    -subj "${SUBJECT}/clientName=${CLIENT}"
+
+openssl x509  -req -in "${CLIENT}.csr" \
+    -extfile <(printf "subjectAltName=DNS:localhost") \
+    -CA ca.crt -CAkey ca.key -out "${CLIENT}.crt" -days "${DAYS}" -sha256 -CAcreateserial
```

Comment:
If you use the same CA for both client and server authentication you have to be _really_ careful about your `extKeyUsage` to avoid malicious actors with user credentials potentially trick other users, or to avoid malicious actors with server credentials potentially use them as client credentials with other servers.